pkg/resource/logging-credentials/logging_operator_secrets.go

```diff
@@ -0,0 +1,50 @@
+package loggingcredentials
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LoggingCredentialsSecretMeta returns metadata for the logging-operator credentials secret.
+func LoggingCredentialsSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      lc.AppConfigName("logging-credentials"),
+		Namespace: "monitoring",
+	}
+}
+
+// Generate a random 20-characters password
+func genPassword() string {
+	const length = 20
+
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789" +
+		"~=+%^*/()[]{}/!@#$?|")
+
+	var b strings.Builder
+	for i := 0; i < length; i++ {
+		b.WriteRune(chars[rand.Intn(len(chars))])
+
+	}
+	str := b.String() // E.g. "ExcbsVQs"
+	return str
+}
+
+// GenerateObservabilityBundleConfigMap returns a configmap for
+// the observabilitybundle application to enable promtail.
+func GenerateLoggingCredentialsBasicSecret(lc loggedcluster.Interface) (v1.Secret, error) {
+
+	secret := v1.Secret{
+		ObjectMeta: LoggingCredentialsSecretMeta(lc),
+		Data:       map[string][]byte{},
```

Comment:
I think you should better use the `DataString` field here to write plain text data, otherwise you have to base64 encode it beforehand.

Reply:
I actually started by base64 encoding my strings, and they ended up being encoded twice. So I got rid of all the extra code and did this, and the created secret is "just enough" base64 encoded :grin: 
But DataString looks neat  :+1: , let me look if it can simplify my code

Reply:
Ok fine, if this works as is.